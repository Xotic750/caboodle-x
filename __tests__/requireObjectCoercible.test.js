import {requireObjectCoercible} from '../dist/caboodle-x';

const hasSymbol = typeof Symbol === 'function' && typeof Symbol('') === 'symbol';
const ifSymbolIt = hasSymbol ? it : xit;

describe('Basic tests', () => {
  it('should throw TypeError everything', () => {
    expect(() => {
      requireObjectCoercible();
    }).toThrow();
    expect(() => {
      requireObjectCoercible(undefined);
    }).toThrow();
    expect(() => {
      requireObjectCoercible(null);
    }).toThrow();
  });

  it('should return value for everything', () => {
    const values = [
      true,
      'abc',
      1,
      function () {},
      [],
      /r/,
    ];
    const actual = values.map(requireObjectCoercible);
    expect(actual).toEqual(values);
  });

  ifSymbolIt('should return Symbol', () => {
    const sym = Symbol('foo');
    expect(requireObjectCoercible(sym)).toBe(sym);
  });
});
