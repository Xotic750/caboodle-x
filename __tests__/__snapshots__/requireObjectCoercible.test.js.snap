// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic tests should throw TypeError everything 1`] = `"Cannot call method on null or undefined"`;

exports[`basic tests should throw TypeError everything 2`] = `"Cannot call method on null or undefined"`;

exports[`basic tests should throw TypeError everything 3`] = `"Cannot call method on null or undefined"`;
